services/src/test/scala/cromwell/pipeline/service/ProjectFileServiceTest.scala

```diff
@@ -54,23 +54,55 @@ class ProjectFileServiceTest extends AsyncWordSpec with Matchers with MockitoSug
+        when(projectVersioning.getUpdatedProjectVersion(project, Some(version)))
+          .thenReturn(Future.successful(Right(version)))
+        when(projectVersioning.updateFile(project, projectFile, version))
+          .thenReturn(Future.successful(Right(updateFiledResponse)))
+        when(projectService.updateProjectVersion(projectId, version, userId)).thenReturn(Future.successful(0))
         projectFileService
           .uploadFile(projectId, projectFile, Some(version), userId)
-          .map(_ shouldBe Right(UpdateFiledResponse(projectFile.path.toString, "master")))
+          .map(_ shouldBe Right(updateFiledResponse))
       }
 
-      "return error message for error request" taggedAs Service in {
-        when(projectVersioning.updateFile(project, projectFile, Some(version)))
+      "Return error message for error request in projectVersioning.getUpdatedProjectVersion" taggedAs Service in {
+        when(projectService.getUserProjectById(projectId, userId)).thenReturn(Future.successful(project))
+        when(projectVersioning.getUpdatedProjectVersion(project, Some(version)))
+          .thenReturn(Future.successful(Left(VersioningException.HttpException("Something wrong"))))
+        projectFileService
+          .uploadFile(projectId, projectFile, Some(version), userId)
+          .map(_ shouldBe Left(VersioningException.HttpException("Something wrong")))
+      }
+
+      "Return error message for error request in projectVersioning.updateFile" taggedAs Service in {
+        when(projectService.getUserProjectById(projectId, userId)).thenReturn(Future.successful(project))
+        when(projectVersioning.getUpdatedProjectVersion(project, Some(version)))
+          .thenReturn(Future.successful(Right(version)))
+        when(projectVersioning.updateFile(project, projectFile, version))
           .thenReturn(Future.successful(Left(VersioningException.HttpException("Something wrong"))))
         projectFileService
           .uploadFile(projectId, projectFile, Some(version), userId)
           .map(_ shouldBe Left(VersioningException.HttpException("Something wrong")))
       }
+
+      "should fail if it is taking too darn long" taggedAs Service in {
+        when(projectService.getUserProjectById(projectId, userId)).thenReturn(Future.successful(project))
+        when(projectVersioning.getUpdatedProjectVersion(project, Some(version)))
+          .thenReturn(Future.successful(Right(version)))
+        when(projectVersioning.updateFile(project, projectFile, version))
+          .thenReturn(Future.successful(Right(updateFiledResponse)))
+        when(projectService.updateProjectVersion(projectId, version, userId)).thenReturn(Future.never)
+        projectFileService
+          .uploadFile(projectId, projectFile, Some(version), userId)
+          .map(
+            _ shouldBe Right(updateFiledResponse)
+          )
+        Thread.sleep(3000)
+        succeed
+      }
```

Comment:
It's an interesting thing to test, but I don't understand what we should do to make this test fail.
You are hardcoding `succeed` case here and I don't see anything in your code that might cause it to fail.